.eslintrc.js

```diff
@@ -328,6 +332,61 @@ module.exports = {
 
     'unused-imports/no-unused-imports': 'error',
     'unused-imports/no-unused-vars': 'off',
+    // New rules added to unicorn
+    'unicorn/no-array-reduce': 'off',
+    'unicorn/no-array-callback-reference': 'off',
+    'unicorn/numeric-separators-style': 'off',
+    'unicorn/prefer-switch': 'off',
+    'unicorn/prefer-spread': 'off',
+    'unicorn/template-indent': 'off',
+    'unicorn/prefer-export-from': 'off',
+    'unicorn/explicit-length-check': 'off',
+    'unicorn/prefer-date-now': 'off',
+    'unicorn/no-await-expression-member': 'off',
+    'unicorn/prefer-dom-node-text-content': 'off',
+    'unicorn/consistent-destructuring': 'off',
+    'unicorn/no-new-array': 'off',
+    'unicorn/prefer-ternary': 'off',
+    'unicorn/prefer-dom-node-dataset': 'off',
+    'unicorn/no-useless-spread': 'off',
+    'unicorn/no-lonely-if': 'off',
+    'unicorn/prefer-array-some': 'off',
+    'unicorn/prefer-native-coercion-functions': 'off',
+    'unicorn/relative-url-style': 'off',
+    'unicorn/prefer-array-flat-map': 'off',
+    'unicorn/prefer-node-protocol': 'off',
+    'unicorn/prefer-module': 'off',
+    'unicorn/no-useless-promise-resolve-reject': 'off',
+    'unicorn/no-array-method-this-argument': 'off',
+    'unicorn/no-array-for-each': 'off',
+    'no-unsafe-optional-chaining': 'off',
+    'unicorn/import-style': 'off',
+    'unicorn/no-array-push-push': 'off',
+    'unicorn/no-useless-switch-case': 'off',
+    'unicorn/prefer-regexp-test': 'off',
+    'unicorn/no-object-as-default-parameter': 'off',
+    'unicorn/prefer-code-point': 'off',
+    'unicorn/prefer-object-from-entries': 'off',
+    'unicorn/prefer-math-trunc': 'off',
+    'unicorn/text-encoding-identifier-case': 'off',
+    'unicorn/no-unreadable-iife': 'off',
+    'unicorn/error-message': 'off',
+    'unicorn/no-thenable': 'off',
+    'unicorn/prefer-array-index-of': 'off',
+
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/no-unnecessary-type-constraint': 'off',
+
+    // End
+
+    // Previous Rules that has been causing errors after upgrading
+    'import/extensions': 'off',
+    'unicorn/consistent-function-scoping': 'off',
+    'unicorn/prevent-abbreviations': 'off',
+    'rxjs/no-nested-subscribe': 'off',
+    'import/no-useless-path-segments': 'off',
+    '@typescript-eslint/restrict-template-expressions': 'off',
+    // End
```

Comment:
Could you provide more context on why **each** rule was disabled? Was it a runtime ESLint error or a significant number of errors in the Sourcegraph codebase? Please share your notes here.

Reply:
These were disabled as a result of errors in the sourcegraph codebase, here are some samples.

1. error  Unexpected use of file extension “schema” for “@sourcegraph/shared/src/schema/settings.schema” in [here](https://github.com/sourcegraph/sourcegraph/blob/main/client/web/src/Layout.tsx#:~:text=import%20%7B%20Settings%20%7D%20from%20%27%40sourcegraph/shared/src/schema/settings.schema%27) and other files too - import/extensions
2. error  The variable `keyFn` should be named `keyFunction`. A more descriptive name will do too  unicorn/prevent-abbreviations in [here](https://github.com/sourcegraph/sourcegraph/blob/main/client/web/src/enterprise/codeintel/badge/hooks/useCodeIntelStatus.ts#:~:text=function%20groupBy%3CV%2C%20K%3E(values%3A%20V%5B%5D%2C%20keyFn%3A%20(value%3A%20V)%20%3D%3E%20K)%3A%20Map%3CK%2C%20V%5B%5D%3E%20%7B) and same type of error in other files too. For these we made the modifications to migrate to such `keyFn` => `keyFunc`, so as to remove disabled rule.
3. error  Nested subscribe calls are forbidden rxjs/no-nested-subscribe in [here](https://github.com/sourcegraph/sourcegraph/blob/main/client/web/src/integration/blob-viewer.test.ts) at line 347 and same type of error in other files too
4. error  Move arrow function to the outer scope  unicorn/consistent-function-scoping in [here](https://github.com/sourcegraph/sourcegraph/blob/main/client/web/src/user/settings/auth/UserSettingsSecurityPage.tsx#:~:text=(accumulator%3A%20AuthProvidersByType%2C%20provider)%20%3D%3E%20%7B) and other files too
5. error  "useThemeProps" is deprecated: Use useTheme hook instead to get theme preference state [here](https://github.com/sourcegraph/sourcegraph/blob/main/client/web/src/Layout.tsx) Line 164:24 - etc/no-deprecated

Reply:
Thanks for the context; it's super helpful!
@gitstart-sourcegraph, for the points `1` and `5`, is it possible to disable these eslint rules for affected [lines](https://github.com/sourcegraph/sourcegraph/blob/main/client/web/src/Layout.tsx#:~:text=import%20%7B%20Settings%20%7D%20from%20%27%40sourcegraph/shared/src/schema/settings.schema%27) [only](https://github.com/sourcegraph/sourcegraph/blob/main/client/web/src/Layout.tsx)?